upf/io/pddl_writer.py

```diff
@@ -182,13 +184,22 @@ def _write_domain(self, out: IO[str]):
                 out.write(f'\n  :precondition (and {" ".join([converter.convert(p) for p in a.preconditions()])})')
             if len(a.effects()) > 0:
                 out.write('\n  :effect (and')
-                for k, v in a.effects():
-                    if v.is_true():
-                        out.write(f' {converter.convert(k)}')
-                    elif v.is_false():
-                        out.write(f' (not {converter.convert(k)})')
+                for e in a.effects():
+                    if e.is_conditional():
+                        out.write(f' (when ({converter.convert(e.condition())}')
```

Comment:
Is it missing a bracket here?
```suggestion
                        out.write(f' (when ({converter.convert(e.condition())})')
```

Reply:
I think it's the opposite, like so:
```suggestion
                        out.write(f' (when {converter.convert(e.condition())}')
```

Because the converter already puts brackets around every expression.

Reply:
Very nice catch btw, you are so meticulous!!